File: frontend/src/app/services/cv-analysis.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CvAnalysisRequest, CvAnalysisResponse } from '../models/cv-analysis.model';

@Injectable({
  providedIn: 'root'
})
export class CvAnalysisService {
  private readonly apiUrl = 'http://localhost:8081/api/cv/analyze';

  constructor(private http: HttpClient) { }

  /**
   * Sends CV text to the backend for analysis
   * @param cvText the CV text to analyze
   * @returns Observable<CvAnalysisResponse> with the analysis results
   */
  analyzeCv(cvText: string): Observable<CvAnalysisResponse> {
    const request: CvAnalysisRequest = { cvText };
    return this.http.post<CvAnalysisResponse>(this.apiUrl, request);
  }
}

File: backend/src/main/java/com/careerflow/cv/service/CvAnalysisService.java
package com.careerflow.cv.service;

import com.careerflow.ai.GeminiCvAnalysisClient;
import com.careerflow.cv.dto.CvAnalysisResponse;
import org.springframework.stereotype.Service;

@Service
public class CvAnalysisService {

    private final GeminiCvAnalysisClient geminiCvAnalysisClient;

    public CvAnalysisService(GeminiCvAnalysisClient geminiCvAnalysisClient) {
        this.geminiCvAnalysisClient = geminiCvAnalysisClient;
    }

    public CvAnalysisResponse analyzeCv(String cvText) {
        return geminiCvAnalysisClient.analyzeCv(cvText);
    }
}